simulation_ws/src/test_nodes/nodes/docking_test.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import rospy
+import rostest
+import time
+import os
+import unittest
+import tf
+
+from math import hypot
+from std_msgs.msg import Bool
+from geometry_msgs.msg import PoseStamped
+from rosgraph_msgs.msg import Clock
+from nav_msgs.msg import Odometry
+from std_srvs.srv import Trigger
+
+from robomaker_simulation_msgs.msg import Tag
+from robomaker_simulation_msgs.srv import Cancel, AddTags
+
+class DockingTest(unittest.TestCase):
+
+    def runTest(self):
+        # Start the docking test
+        self.test_docking()
+
+    def setUp(self):
+        self.test_name = "Robot_Monitoring_Tests_" + \
+            str(time.time()).split(".", 1)[0]
+        rospy.loginfo("Test Name: %s", self.test_name)
+        self.is_docked = False
+        self.is_docking = False
+        self.get_params()
+        while rospy.Time.now().secs == 0:
+            pass
+        self.starting_time_secs = rospy.Time.now().secs
+        self.timer = rospy.Timer(30, callback=self.timer_cb(), oneshot=True)
+
+    def timer_cb(self):
+        if self.trigger_docking_client():
+            self.is_docking = True
+
+    def trigger_docking_client(self):
+        rospy.wait_for_service('trigger_docking_service')
+        try:
+            client_trigger = rospy.ServiceProxy('trigger_docking_service', Trigger)
+            response = client_trigger()
```

Comment:
Tag the "going to dock" timestamp, so we can measure how long the robot ran to go dock